railties/lib/rails/generators/app_base.rb

```diff
@@ -312,6 +324,52 @@ def gem_for_database
         end
       end
 
+      def generate_purpose_model
+        return unless options[:purposeful] == 'true'
+
+        `mkdir app`
+        `mkdir app/models`
+        `touch app/models/purpose.rb`
+
+        File.open('app/models/purpose.rb', 'w') do |file|
+          file.write(
+            <<~HEREDOC
+              class Purpose < ActiveYaml::Base
+                include ActiveHash::Associations
+
+                field :name
+
+                # a purpose can have many children_purposes (cf. purpose_tree)
+                has_many :children, class_name: 'Purpose'
```

Comment:
does this workd @olegsfinest? is it even possible to model this kind of relation using the active_yaml file?
what we want of course is that children purposes in the yaml file are automatically created as related model entries....